File: Backend/fairshareapi/src/main/java/com/priyanshparekh/fairshareapi/expense/Expense.java
package com.priyanshparekh.fairshareapi.expense;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.regex.Pattern;

@Setter
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "expenses")
public class Expense {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long groupId;
    private String name;
    private Double amount;
    private String note;
    private Long paidBy;
    private Long createdAt;
    private String receiptUrl;

}

File: Frontend/FairShare/app/src/main/java/com/priyanshparekh/fairshare/model/Expense.kt
package com.priyanshparekh.fairshare.model

data class Expense(
    val id: Long?,
    val groupId: Long?,
    val name: String,
    val amount: Double,
    val note: String,
    val paidBy: Long,
    val splitBetween: List<User>,
    val createdAt: Long,
    var receiptUrl: String?
) {

    override fun toString(): String {
        return "Group id: $groupId, Name: $name, Amount: $amount, Note: $note, Paid By: $paidBy, Split Between: ${splitBetween.size}, Date: $createdAt"
    }
}